### PR TITLE
Issue #3100: Added sorts to ArticleList sql where it seemed necessary.

### PR DESCRIPTION
**Pull request by stefanhaerter**
Extended cache key for ArticleList.

Referencing #3100 and closing #3424

### CODE REVIEW

File: Kernel/System/Ticket/Article.pm [hunk 1]
```diff
@@ -1197,17 +1197,21 @@ sub _MetaArticleList {
         return;
     }
 
-    my $CacheKey = '_MetaArticleList::' . $Param{TicketID};
+    my $CacheKey
```
Comment (bschmalhofer) [inline Kernel/System/Ticket/Article.pm:1203]: Just a bit nitpicking. For VersionView and ShowDeletedArticles any true value could be passed. But the later cache clearing code only clears for the values 0 and 1. This edge case could be avoided by using dedicated variables 
`my $VersionView = $Param{VersionView} ? 1 : 0;`

Comment (stefanhaerter) [reply Kernel/System/Ticket/Article.pm:1203]: I intended to handle this with `int( $Param{VersionView} )` - is this not sufficient? And if so, why?

Comment (bschmalhofer) [reply Kernel/System/Ticket/Article.pm:1203]: If hypothetically `VersionView => 99.9` is passed, than int('99.9') = 99 would be used as Key for getting and setting the cache. But 1 would still be the Key for clearing the cache.

Comment (stefanhaerter) [reply Kernel/System/Ticket/Article.pm:1203]: I see - you're right, as usual ;) I'll change this

**Review by bschmalhofer — APPROVED** [review]
Looks good to me.

**Pull request merged**